hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -187,23 +189,46 @@ private[sql] object SchemaConverters {
           .values(toAvroType(vt, valueContainsNull, recordName, nameSpace))
       case st: StructType =>
         val childNameSpace = if (nameSpace != "") s"$nameSpace.$recordName" else recordName
-        val fieldsAssembler = builder.record(recordName).namespace(nameSpace).fields()
-        st.foreach { f =>
-          val fieldAvroType =
-            toAvroType(f.dataType, f.nullable, f.name, childNameSpace)
-          fieldsAssembler.name(f.name).`type`(fieldAvroType).noDefault()
+        if (canBeUnion(st)) {
```

Comment:
Are these from Spark?  How do we track any compatibility issues?

Reply:
This is our own. Not sure i understood your other q regarding compatibility issues? Can you elaborate?

Reply:
I see this is in spark-common, so it is applied for all Spark versions.  Want to see if there's any chance that this can be broken in a particular version.

Reply:
We have test for it that we run across all the versions we're claiming to be compatible with.

Reply:
Sg